pkg/helm/chartproxy/proxy.go

```diff
@@ -93,29 +93,34 @@ func (p *proxy) IndexFile(onlyCompatible bool, namespace string) (*repo.IndexFil
 				continue
 			}
 
-			for key, entry := range idxFile.Entries {
-				for i := len(entry) - 1; i >= 0; i-- {
-					if entry[i].Type == "library" {
-						entry = append(entry[:i], entry[i+1:]...)
+			for key, entries := range idxFile.Entries {
+				for i := len(entries) - 1; i >= 0; i-- {
+					if entries[i] == nil || entries[i].Metadata == nil {
+						klog.Warningf("Helm chart %v from repository %v has an invalid entry at index %v", key, helmRepo.Name, i)
+						entries = append(entries[:i], entries[i+1:]...)
 						continue
 					}
```

Comment:
This is actually the 5 new lines.

The rest is only because I renamed entry to entries because it's an array.

Reply:
/cc @Kartikey-star @jhadvig